Source/Fuse.Elements/Resources/Exif/Exif.uno

```
@@ -0,0 +1,190 @@
+
+			if (imageSource) {
+				CGImageMetadataRef metadata = CGImageSourceCopyMetadataAtIndex(imageSource, 0, NULL);
+				if (metadata) {
+					metadataArray = CFBridgingRelease(CGImageMetadataCopyTags(metadata));
+					CFRelease(metadata);
+				}
+				CFRelease(imageSource);
+			}
+
+			int rotation = 0;
+			NSString* tagTarget = @"Orientation";
+			if (metadataArray != nil)
+			{
+				for (id tag in metadataArray) {
+					CFStringRef tagName = CGImageMetadataTagCopyName((CGImageMetadataTagRef)tag);
+					if ([tagTarget isEqualToString: (__bridge NSString*)tagName]) {
+						CFTypeRef rot = CGImageMetadataTagCopyValue((CGImageMetadataTagRef)tag);
+						rotation = [((__bridge NSNumber*)rot) intValue];
+						break;
+					}
+				}
+			}
+
+			return rotation;
+
+		@}
+	}
+
+	[Require("Gradle.Dependency.Compile", "com.drewnoakes:metadata-extractor:2.10.1")]
+	[ForeignInclude(Language.Java,
+		"java.io.ByteArrayInputStream",
+		"java.io.InputStream",
+		"java.io.IOException",
+		"com.drew.imaging.ImageMetadataReader",
+		"com.drew.imaging.ImageProcessingException",
+		"com.drew.metadata.Metadata",
+		"com.drew.metadata.MetadataException",
+		"com.drew.metadata.exif.ExifIFD0Directory")]
+	//Uses https://drewnoakes.com/code/exif/
+	extern(Android) class ExifAndroidImpl
+	{
+		internal static ExifData FromByteArray(byte[] bytes)
+		{
+			var buf = ForeignDataView.Create(bytes);
+			var orientation = GetOrientation(buf);
+			return new ExifData(orientation);
+		}
+
+		[Foreign(Language.Java)]
+		static int GetOrientation(Java.Object buf)
+		@{
+			InputStream s = new com.fuse.android.ByteBufferInputStream((com.uno.UnoBackedByteBuffer)buf);
+			try {
+				Metadata metadata = ImageMetadataReader.readMetadata(s);
+				ExifIFD0Directory directory = metadata.getFirstDirectoryOfType(ExifIFD0Directory.class);
+
+				if (directory != null && directory.containsTag(ExifIFD0Directory.TAG_ORIENTATION)) {
+					return directory.getInt(ExifIFD0Directory.TAG_ORIENTATION);
```

I think this still risks throwing in the case where `ExifIFD0Directory.TAG_ORIENTATION` isn't an int. But I guess that's such a bizarre case, that it's not worth worrying about. We'd just let it throw. In other words, this looks like a nice compromise.